common/version.go

```diff
@@ -6,23 +6,39 @@ import (
 	pbcommon "github.com/drand/drand/protobuf/common"
 )
 
-const (
-	FallbackMajor = 0
-	FallbackMinor = 0
-	FallbackPatch = 0
+// Must be manually updated!
+// Before releasing: Verify the version number and set Prerelease to ""
+// After releasing: Increase the Patch number and set Prerelease to "-pre"
+var version = Version{
+	Major:      1,
+	Minor:      1,
+	Patch:      2,
+	Prerelease: "+pre",
+}
+
+// Set via -ldflags. Example:
+//   go install -ldflags "-X common.BUILDDATE=`date -u +%d/%m/%Y@%H:%M:%S` -X common.GITCOMMIT=`git rev-parse HEAD`
+//
+// See the Makefile and the Dockerfile in the root directory of the repo
+var (
+	COMMIT    = ""
+	BUILDDATE = ""
 )
 
+func GetAppVersion() Version {
+	return version
+}
+
 type Version struct {
-	Major uint32
-	Minor uint32
-	Patch uint32
+	Major      uint32
+	Minor      uint32
+	Patch      uint32
+	Prerelease string
 }
 
 func (v Version) IsCompatible(verRcv Version) bool {
-	if verRcv.Major == FallbackMajor && verRcv.Minor == FallbackMinor && verRcv.Patch == FallbackPatch {
-		return true
-	}
-	if v.Major == verRcv.Major {
+	// TODO Not sure what we should do here
+	if verRcv.Prerelease == "" || v.Major == verRcv.Major {
```

Comment:
This seems a bit too permissive: we don't have any promises of forward compatibility on releases, do we?
I guess we should match `Major` and `Minor`.

Are we sure that the current binary (master tip) is retro-compatible with the `v1.3.1`? 
Or do we plan to release it as a `v2.0.0`? @willscott  @nikkolasg

Reply:
yes, that's why I put in that TODO... I'll check `Major` and `Minor` and wait for @willscott / @nikkolasg to shed some light.